### PR TITLE
feat(v2): blog sidebar

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Feature parity with Docusaurus v1.

https://docusaurus.io/docs/en/adding-blog#changing-how-many-blog-posts-show-on-sidebar

Needed for RN migration, cc @Simek 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

tests + preview + dogfooding

### CODE REVIEW

**Comment by slorber** [comment]
@lex111 any opinion on the design we should use for this feature? (I'm not very good at design)

Here's a POC, will polish this a bit to before merge and cleanup TOC classes I reused: 
https://deploy-preview-3593--docusaurus-2.netlify.app/classic/blog/2019/12/30/docusaurus-2019-recap

**Comment by lex111** [comment]
@slorber definitely, the recent posts section should **not** look like TOC.

**Comment by slorber** [comment]
@lex111 any design suggestion for this feature then?

Maybe you want it to be more like v1? (no border etc...)
https://prettier.io/blog/2020/08/24/2.1.0.html

I'll polish the node part but if you have a good idea don't hesitate to modify the BlogSidebar UI component, as all the design is basically there we shouldn't have any conflict

**Comment by lex111** [comment]
> Maybe you want it to be more like v1? (no border etc...)

Yes, that's what I meant. Let's remove the border, make the fonts bigger. You can look at v1 as an example https://docusaurus.io/blog/2019/12/30/docusaurus-2019-recap

**Comment by slorber** [comment]
@lex111 does it look good enough to merge?

File: packages/docusaurus-theme-classic/src/theme/BlogSidebar/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.sidebarItemTitle {
+  margin-bottom: 0.5rem;
+}
+
+.sidebarItemList {
+  overflow-y: auto;
+  font-size: 0.8rem;
+  padding-left: 0.3rem;
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/BlogSidebar/styles.module.css:23]: Maybe make the font a little bigger?

```suggestion
  font-size: .9rem;
  padding-left: 0;
```

| Before   | After    |
| -------- | -------- |
| image | image |

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/BlogSidebar/styles.module.css:23]: Why not. Ready to merge with your suggestion?

File: packages/docusaurus-theme-classic/src/theme/BlogListPage/index.tsx [hunk 2]
```diff
@@ -25,7 +26,10 @@ function BlogListPage(props: Props): JSX.Element {
     <Layout title={title} description={blogDescription}>
       <div className="container margin-vert--lg">
         <div className="row">
-          <main className="col col--8 col--offset-2">
+          <div className="col col--2">
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/BlogListPage/index.tsx:29]: Not a fan of empty divs (if `blogSidebarCount=0`), let's handle this case properly (add `col--offset-2` class if there is no sidebar).

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/BlogListPage/index.tsx:29]: The problem is that it requires to do this check everywhere 🤪 
I'd rather encapsulate it and keep a constant layout, makes code simpler for same results.
I don't think it's a bad practice to put an empty div for layout reasons, particularly if it has a width. It's common with flexbox to do so.

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/BlogListPage/index.tsx:29]: Well, okay, although I don't really like this approach.

**Review by lex111 — APPROVED** [review]
Feel free to self-merge.

**Comment by lex111** [comment]
Woohoo, we get 3k commits! :D

image

**Comment by slorber** [comment]
Nice 😉soon 20k stars

**Pull request merged**